src/winmain.cpp

```diff
@@ -590,6 +589,70 @@ LRESULT CALLBACK proxyDlgProc(HWND hWndDlg, UINT Msg, WPARAM wParam, LPARAM)
 	return FALSE;
 }
 
+struct UpdateCheckParams
+{
+	GupNativeLang& _nativeLang;
+	GupParameters& _gupParams;
+};
+
+LRESULT CALLBACK updateCheckDlgProc(HWND hWndDlg, UINT message, WPARAM wParam, LPARAM lParam)
+{
+	switch (message)
+	{
+	case WM_INITDIALOG:
+	{
+		auto* params = reinterpret_cast<UpdateCheckParams*>(lParam);
+		if (params)
+		{
+			const string& title = params->_gupParams.getMessageBoxTitle();
+			if (!title.empty())
+				::SetWindowTextA(hWndDlg, title.c_str());
+			string text1 = params->_nativeLang.getMessageString("MSGID_NOUPDATE1");
+			if (!text1.empty())
+				::SetDlgItemTextA(hWndDlg, IDC_UPDATE_STATIC1, text1.c_str());
+			string text2 = params->_nativeLang.getMessageString("MSGID_NOUPDATE2");
+			if (!text2.empty())
+				::SetDlgItemTextA(hWndDlg, IDC_UPDATE_STATIC2, text2.c_str());
+		}
+		goToScreenCenter(hWndDlg);
+		return TRUE;
+	}
+
+	case WM_COMMAND:
+	{
+		switch LOWORD((wParam))
+		{
+		case IDOK:
+		case IDYES:
+		case IDNO:
+		case IDCANCEL:
+			EndDialog(hWndDlg, wParam);
+			return TRUE;
+		default:
+			break;
+		}
+	}
+
+	case WM_NOTIFY:
+		switch (((LPNMHDR)lParam)->code)
+		{
+		case NM_CLICK:
+		case NM_RETURN:
+		{
+			PNMLINK pNMLink = (PNMLINK)lParam;
+			LITEM item = pNMLink->item;
+			if (lstrcmp(item.szID, TEXT("id_download")) == 0)

```

Comment:
Done. 

I don't have MSVC 2015 so I had to convert the project to MSVC 2017 locally.
Because of that, it got converted to Unicode instead of Multi-Byte.

Should WinGUP be updated to MSVC 2017/2019 at some point?

Reply:
WinGUp is already under MSVC 2017

Reply:
Hm... for some reason when I opened in MSVC 2017, it asked me to upgrade the project.

Reply:
You're right. WinGUp is under MSVC 2015.
But in my build machine it's updated to VC 2017, and it's built with VC 2017